source/dofs/dof_handler_policy.cc

```diff
@@ -1363,25 +2044,53 @@ namespace internal
                                                           cell,
                                                           next_free_dof);
 
-          // Step 2: unify dof indices in case this is an hp DoFHandler
-          //
-          // during unification, we need to renumber DoF indices. there,
-          // we can check that all previous DoF indices were valid, but
-          // this only makes sense if we really distributed DoFs on
-          // all (non-artificial) cells above
-          next_free_dof =
-            unify_dof_indices(dof_handler,
-                              next_free_dof,
-                              /* check_validity = */
-                              (subdomain_id == numbers::invalid_subdomain_id));
-
           update_all_active_cell_dof_indices_caches(dof_handler);
 
           return next_free_dof;
         }
 
 
 
+        /**
+         * During DoF distribution, DoFs on ghost interfaces get different
+         * indices assigned by each adjacent subdomain. We need to clarify
+         * ownership of those DoFs by imposing a criterion, which is that
+         * the adjacent subdomain belonging to the processor of lowest rank
+         * prescribes its index.
+         *
+         * Thus, we have to invalidate all those DoFs on ghost cells that
+         * belong to processors of lower rank than the current one. Later
+         * during ghost exchange in Phase 5, these values will be overwritten
```

Comment:
Should this be phase 4 (the first ghost exchange)? Phase 4 should take care of everything on locally owned/ghost interfaces. Phase 5 is only necessary at the far edges of ghost cells.

Reply:
I renumbered all phases to be in accordance with the distributed paper (see lines 4851-4865). 'Phase 5' now relates to the first ghost exchange.